### PR TITLE
google_sql_database_instance - support for secondary_zone

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes [#10663](https://github.com/hashicorp/terraform-provider-google/issues/10663)

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
sql: added `settings.location_preference.secondary_zone` field in `google_sql_database_instance`
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb [hunk 1]
```diff
@@ -1070,6 +1075,7 @@ func expandLocationPreference(configured []interface{}) *sqladmin.LocationPrefer
 	return &sqladmin.LocationPreference{
 		FollowGaeApplication: _locationPreference["follow_gae_application"].(string),
 		Zone:                 _locationPreference["zone"].(string),
+	    SecondaryZone:        _locationPreference["secondary_zone"].(string),
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb:1078]: nit: I think the spacing here is inconsistent- can you ensure it's using tabs and lines up w/ above?

**Review by rileykarson — COMMENTED** [review]
One minor nitpick on spacing, otherwise LGTM!

File: mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb (hunk 1 above)
Comment (DrFaust92) [reply mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb]: fixed!

**Comment by rileykarson** [comment]
The VCR failures should be addressed with https://github.com/GoogleCloudPlatform/magic-modules/pull/6189. Do you mind removing draft mode by marking the PR ready to review?

**Comment by DrFaust92** [comment]
Thanks [rileykarson](https://github.com/rileykarson), should i rebase?

**Review by rileykarson — APPROVED** [review]
Nope, you're all good!

**Pull request merged**